src/main/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaFeedbackSuggestionsService.java

```diff
@@ -0,0 +1,103 @@
+package de.tum.in.www1.artemis.service.connectors.athena;
+
+import java.util.List;
+
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import de.tum.in.www1.artemis.domain.GradingInstruction;
+import de.tum.in.www1.artemis.domain.TextBlockRef;
+import de.tum.in.www1.artemis.domain.TextExercise;
+import de.tum.in.www1.artemis.domain.TextSubmission;
+import de.tum.in.www1.artemis.domain.enumeration.FeedbackType;
+import de.tum.in.www1.artemis.exception.NetworkingError;
+import de.tum.in.www1.artemis.repository.GradingInstructionRepository;
+import de.tum.in.www1.artemis.service.dto.athena.TextExerciseDTO;
+import de.tum.in.www1.artemis.service.dto.athena.TextFeedbackDTO;
+import de.tum.in.www1.artemis.service.dto.athena.TextSubmissionDTO;
+
+/**
+ * Service for receiving feedback suggestions from the Athena service.
+ * Assumes that submissions and already given feedback have already been sent to Athena.
+ */
+@Service
+@Profile("athena")
+public class AthenaFeedbackSuggestionsService {
+
+    private final Logger log = LoggerFactory.getLogger(AthenaFeedbackSuggestionsService.class);
+
+    @Value("${artemis.athena.url}")
+    private String athenaUrl;
```

Comment:
I would prefer to leave this as-is because most configuration value injections in Artemis are handled like this. What do you think?

Reply:
While I favor constructor injection and prefer making fields final, I see your point. To be honest, I think we just use field injection in most places because initially it might not have been supported or people didn't know this was possible and others just adapted to what was already there. So, if we want to use this, we should probably open a separate PR that unifies this for all occurrences.